lib/logstash/inputs/syslog.rb

```diff
@@ -138,7 +138,11 @@ def udp_listener(output_queue)
     @logger.info("Starting syslog udp listener", :address => "#{@host}:#{@port}")
 
     @udp.close if @udp
-    @udp = UDPSocket.new(Socket::AF_INET)
+    if IPAddr.new(@host).ipv6?
+      @udp = UDPSocket.new(Socket::AF_INET6)
+    elsif IPAddr.new(@host).ipv4?
+      @udp = UDPSocket.new(Socket::AF_INET)
+    end
```

Comment:
believe this breaks backwards compatibility in cases where host isn't an ip address.
some users might have used `host => ''` - previously we would reach `bind('', 12345)`.
in cases such as those we'll get an `IPAddr::InvalidAddressError` with the PR's changes.

```suggestion
    @udp = UDPSocket.new (IPAddr.new(@host).ipv6? rescue nil) ? Socket::AF_INET6 : Socket::AF_INET
```

Reply:
Great input, I didn't think about that use-case so thanks for your wisdom on this subject :-)